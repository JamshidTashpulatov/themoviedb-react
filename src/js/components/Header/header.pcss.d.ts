export const header: string;
export const navigation: string;
export const menu: string;
export const menuitem: string;
export const menulink: string;
export const active: string;
export const button: string;
