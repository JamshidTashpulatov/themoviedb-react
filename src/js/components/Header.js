import React from 'react';
import { IndexLink } from 'react-router';
import BackButton from './BackButton';
import Search from './Search';
import SearchBox from './SearchBox';
import Styles from '../../css/header.pcss';

class Header extends React.Component {
  constructor() {
    super();
    this.toggleSearchBox = this.toggleSearchBox.bind(this);
    this.state = {
      toggleSearchBox: false,
    }
  }
  toggleSearchBox() {
    this.setState({ toggleSearchBox: this.state.toggleSearchBox ? false : true });
  }
  render() {

    return (
      <div className="header-container">
        <div className={Styles.header}>
          <div className={Styles.navigation}>
            <BackButton goBack={this.props.history.goBack} />
            <ul className={Styles.menu}>
              <li className={Styles.menulink}><IndexLink activeClassName="active" to='/'>Home</IndexLink></li>
              <li className={Styles.menulink}><IndexLink activeClassName="active" to='/about'>About</IndexLink></li>
            </ul>
            <Search toggleSearchBox={this.toggleSearchBox}/>
          </div>
        </div>
        <SearchBox collapsed={this.state.toggleSearchBox}/>
      </div>
    );
  }
}

export default Header
