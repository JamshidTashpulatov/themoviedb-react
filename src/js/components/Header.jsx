import React from 'react';
import { Link } from 'react-router-dom';
import SearchBox from './SearchBox';
import Styles from '../../css/header.pcss';
import { withRouter } from 'react-router-dom';

class Header extends React.Component {
  constructor() {
    super();
    this.toggleSearchBox = this.toggleSearchBox.bind(this);
    this.state = {
      toggleSearchBox: false,
    };
  }
  toggleSearchBox() {
    this.setState({ toggleSearchBox: this.state.toggleSearchBox === false });
  }
  render() {
    return (
      <div className="header-container">
        <div className={Styles.header}>
          <div className={Styles.navigation}>
            <button className={`${Styles.button} material-icons`} onClick={this.props.history.goBack}>arrow_back</button>
            <ul className={Styles.menu}>
              <li className={Styles.menulink}>
                <Link activeClassName="active" to="/">Latest</Link>
              </li>
              <li className={Styles.menulink}>
                <Link activeClassName="active" to="/genres/all">Genres</Link>
              </li>
              <li className={Styles.menulink}>
                <Link activeClassName="active" to="/about">About</Link>
              </li>
            </ul>
            <button className={`${Styles.button} material-icons`} onClick={this.toggleSearchBox}>search</button>
          </div>
        </div>
        <SearchBox collapsed={this.state.toggleSearchBox} toggleSearchBox={this.toggleSearchBox} />
      </div>
    );
  }
}

Header.propTypes = {
  history: React.PropTypes.object, // eslint-disable-line react/forbid-prop-types
};

Header.defaultProps = {
  history: {},
};

export default withRouter(Header);
