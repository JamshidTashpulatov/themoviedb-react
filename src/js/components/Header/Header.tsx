import * as React from "react";
import { NavLink } from 'react-router-dom';
import SearchBox from '../SearchBox';
import Styles from './header.pcss';
import { withRouter } from 'react-router-dom';
type HeaderProps = {
  history?: object
};
type HeaderState = {
  toggleSearchBox: boolean
};
class Header extends React.Component<HeaderProps, HeaderState> {
  constructor(props) {
    super(props);
    this.toggleSearchBox = this.toggleSearchBox.bind(this);
    this.state = {
      toggleSearchBox: false
    };
  }
  toggleSearchBox() {
    this.setState({ toggleSearchBox: this.state.toggleSearchBox === false });
  }
  render() {
    return (
      <div>
        <div className={Styles.header}>
          <div className={Styles.navigation}>
            <button className={`${Styles.button} material-icons`} onClick={this.props.history.goBack}>
              arrow_back
            </button>
            <ul className={Styles.menu}>
              <li className={Styles.menuitem}>
                <NavLink exact className={Styles.menulink} activeClassName={Styles.active} to="/">
                  Latest
                </NavLink>
              </li>
              <li className={Styles.menuitem}>
                <NavLink className={Styles.menulink} activeClassName={Styles.active} to="/genres">
                  Genres
                </NavLink>
              </li>
              <li className={Styles.menuitem}>
                <NavLink className={Styles.menulink} activeClassName={Styles.active} to="/about">
                  About
                </NavLink>
              </li>
            </ul>
            <button className={`${Styles.button} material-icons`} onClick={this.toggleSearchBox}>
              search
            </button>
          </div>
        </div>
        <SearchBox collapsed={this.state.toggleSearchBox} toggleSearchBox={this.toggleSearchBox} />
      </div>
    );
  }
}
Header.defaultProps = {
  history: {}
};
export default withRouter(Header);
