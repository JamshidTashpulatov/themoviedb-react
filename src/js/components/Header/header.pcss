@import "../../../css/variables.pcss";

.header {
  position: fixed;
  width: 100%;
  top: 0;
  background: var(--bgalt);
  z-index: 100;
}

.navigation {
  display: flex;
  justify-content: space-between;
  line-height: 40px;
  font-size: 16px;

  @media (--desktop) {
    font-size: 28px !important;
    line-height: 50px;
  }
}

.menu {
  text-align: center;
  display: flex;
  margin: 0;
}

.menuitem {
  padding: 0 15px;
  font-weight: 600;
  list-style-type: none;
}

.menulink {
  color: #fff;
  text-decoration: none;
  text-transform: uppercase;
}

.active {
  color: var(--fontcolor);
}

.button {
  cursor: pointer;
  background: transparent;
  border: none;
  outline: none;
  color: #fff;
  font-size: 20px !important;

  @media (--desktop) {
    font-size: 28px !important;
  }
}
