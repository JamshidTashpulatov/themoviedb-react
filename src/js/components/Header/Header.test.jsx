import React from "react";
import { shallow, mount, render } from "enzyme";
import Header from "./Header";

import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

it("renders correctly", () => {
  const tree = shallow(<Header.WrappedComponent />);

  expect(tree).toMatchSnapshot();
});
