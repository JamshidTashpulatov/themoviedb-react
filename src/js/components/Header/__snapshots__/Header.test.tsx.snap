// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header
    history={Object {}}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="header"
        >
          <div
            className="navigation"
          >
            <button
              className="button material-icons"
              onClick={undefined}
            >
              arrow_back
            </button>
            <ul
              className="menu"
            >
              <li
                className="menuitem"
              >
                <NavLink
                  activeClassName="active"
                  ariaCurrent="true"
                  className="menulink"
                  exact={true}
                  to="/"
                >
                  Latest
                </NavLink>
              </li>
              <li
                className="menuitem"
              >
                <NavLink
                  activeClassName="active"
                  ariaCurrent="true"
                  className="menulink"
                  to="/genres"
                >
                  Genres
                </NavLink>
              </li>
              <li
                className="menuitem"
              >
                <NavLink
                  activeClassName="active"
                  ariaCurrent="true"
                  className="menulink"
                  to="/about"
                >
                  About
                </NavLink>
              </li>
            </ul>
            <button
              className="button material-icons"
              onClick={[Function]}
            >
              search
            </button>
          </div>
        </div>,
        <Connect(OnClickOutside(SearchBox))
          collapsed={false}
          toggleSearchBox={[Function]}
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="navigation"
          >
            <button
              className="button material-icons"
              onClick={undefined}
            >
              arrow_back
            </button>
            <ul
              className="menu"
            >
              <li
                className="menuitem"
              >
                <NavLink
                  activeClassName="active"
                  ariaCurrent="true"
                  className="menulink"
                  exact={true}
                  to="/"
                >
                  Latest
                </NavLink>
              </li>
              <li
                className="menuitem"
              >
                <NavLink
                  activeClassName="active"
                  ariaCurrent="true"
                  className="menulink"
                  to="/genres"
                >
                  Genres
                </NavLink>
              </li>
              <li
                className="menuitem"
              >
                <NavLink
                  activeClassName="active"
                  ariaCurrent="true"
                  className="menulink"
                  to="/about"
                >
                  About
                </NavLink>
              </li>
            </ul>
            <button
              className="button material-icons"
              onClick={[Function]}
            >
              search
            </button>
          </div>,
          "className": "header",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <button
                className="button material-icons"
                onClick={undefined}
              >
                arrow_back
              </button>,
              <ul
                className="menu"
              >
                <li
                  className="menuitem"
                >
                  <NavLink
                    activeClassName="active"
                    ariaCurrent="true"
                    className="menulink"
                    exact={true}
                    to="/"
                  >
                    Latest
                  </NavLink>
                </li>
                <li
                  className="menuitem"
                >
                  <NavLink
                    activeClassName="active"
                    ariaCurrent="true"
                    className="menulink"
                    to="/genres"
                  >
                    Genres
                  </NavLink>
                </li>
                <li
                  className="menuitem"
                >
                  <NavLink
                    activeClassName="active"
                    ariaCurrent="true"
                    className="menulink"
                    to="/about"
                  >
                    About
                  </NavLink>
                </li>
              </ul>,
              <button
                className="button material-icons"
                onClick={[Function]}
              >
                search
              </button>,
            ],
            "className": "navigation",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "arrow_back",
                "className": "button material-icons",
                "onClick": undefined,
              },
              "ref": null,
              "rendered": "arrow_back",
              "type": "button",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <li
                    className="menuitem"
                  >
                    <NavLink
                      activeClassName="active"
                      ariaCurrent="true"
                      className="menulink"
                      exact={true}
                      to="/"
                    >
                      Latest
                    </NavLink>
                  </li>,
                  <li
                    className="menuitem"
                  >
                    <NavLink
                      activeClassName="active"
                      ariaCurrent="true"
                      className="menulink"
                      to="/genres"
                    >
                      Genres
                    </NavLink>
                  </li>,
                  <li
                    className="menuitem"
                  >
                    <NavLink
                      activeClassName="active"
                      ariaCurrent="true"
                      className="menulink"
                      to="/about"
                    >
                      About
                    </NavLink>
                  </li>,
                ],
                "className": "menu",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <NavLink
                      activeClassName="active"
                      ariaCurrent="true"
                      className="menulink"
                      exact={true}
                      to="/"
                    >
                      Latest
                    </NavLink>,
                    "className": "menuitem",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "activeClassName": "active",
                      "ariaCurrent": "true",
                      "children": "Latest",
                      "className": "menulink",
                      "exact": true,
                      "to": "/",
                    },
                    "ref": null,
                    "rendered": "Latest",
                    "type": [Function],
                  },
                  "type": "li",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <NavLink
                      activeClassName="active"
                      ariaCurrent="true"
                      className="menulink"
                      to="/genres"
                    >
                      Genres
                    </NavLink>,
                    "className": "menuitem",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "activeClassName": "active",
                      "ariaCurrent": "true",
                      "children": "Genres",
                      "className": "menulink",
                      "to": "/genres",
                    },
                    "ref": null,
                    "rendered": "Genres",
                    "type": [Function],
                  },
                  "type": "li",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <NavLink
                      activeClassName="active"
                      ariaCurrent="true"
                      className="menulink"
                      to="/about"
                    >
                      About
                    </NavLink>,
                    "className": "menuitem",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "activeClassName": "active",
                      "ariaCurrent": "true",
                      "children": "About",
                      "className": "menulink",
                      "to": "/about",
                    },
                    "ref": null,
                    "rendered": "About",
                    "type": [Function],
                  },
                  "type": "li",
                },
              ],
              "type": "ul",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "search",
                "className": "button material-icons",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "search",
              "type": "button",
            },
          ],
          "type": "div",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "collapsed": false,
          "toggleSearchBox": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="header"
          >
            <div
              className="navigation"
            >
              <button
                className="button material-icons"
                onClick={undefined}
              >
                arrow_back
              </button>
              <ul
                className="menu"
              >
                <li
                  className="menuitem"
                >
                  <NavLink
                    activeClassName="active"
                    ariaCurrent="true"
                    className="menulink"
                    exact={true}
                    to="/"
                  >
                    Latest
                  </NavLink>
                </li>
                <li
                  className="menuitem"
                >
                  <NavLink
                    activeClassName="active"
                    ariaCurrent="true"
                    className="menulink"
                    to="/genres"
                  >
                    Genres
                  </NavLink>
                </li>
                <li
                  className="menuitem"
                >
                  <NavLink
                    activeClassName="active"
                    ariaCurrent="true"
                    className="menulink"
                    to="/about"
                  >
                    About
                  </NavLink>
                </li>
              </ul>
              <button
                className="button material-icons"
                onClick={[Function]}
              >
                search
              </button>
            </div>
          </div>,
          <Connect(OnClickOutside(SearchBox))
            collapsed={false}
            toggleSearchBox={[Function]}
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="navigation"
            >
              <button
                className="button material-icons"
                onClick={undefined}
              >
                arrow_back
              </button>
              <ul
                className="menu"
              >
                <li
                  className="menuitem"
                >
                  <NavLink
                    activeClassName="active"
                    ariaCurrent="true"
                    className="menulink"
                    exact={true}
                    to="/"
                  >
                    Latest
                  </NavLink>
                </li>
                <li
                  className="menuitem"
                >
                  <NavLink
                    activeClassName="active"
                    ariaCurrent="true"
                    className="menulink"
                    to="/genres"
                  >
                    Genres
                  </NavLink>
                </li>
                <li
                  className="menuitem"
                >
                  <NavLink
                    activeClassName="active"
                    ariaCurrent="true"
                    className="menulink"
                    to="/about"
                  >
                    About
                  </NavLink>
                </li>
              </ul>
              <button
                className="button material-icons"
                onClick={[Function]}
              >
                search
              </button>
            </div>,
            "className": "header",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <button
                  className="button material-icons"
                  onClick={undefined}
                >
                  arrow_back
                </button>,
                <ul
                  className="menu"
                >
                  <li
                    className="menuitem"
                  >
                    <NavLink
                      activeClassName="active"
                      ariaCurrent="true"
                      className="menulink"
                      exact={true}
                      to="/"
                    >
                      Latest
                    </NavLink>
                  </li>
                  <li
                    className="menuitem"
                  >
                    <NavLink
                      activeClassName="active"
                      ariaCurrent="true"
                      className="menulink"
                      to="/genres"
                    >
                      Genres
                    </NavLink>
                  </li>
                  <li
                    className="menuitem"
                  >
                    <NavLink
                      activeClassName="active"
                      ariaCurrent="true"
                      className="menulink"
                      to="/about"
                    >
                      About
                    </NavLink>
                  </li>
                </ul>,
                <button
                  className="button material-icons"
                  onClick={[Function]}
                >
                  search
                </button>,
              ],
              "className": "navigation",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "arrow_back",
                  "className": "button material-icons",
                  "onClick": undefined,
                },
                "ref": null,
                "rendered": "arrow_back",
                "type": "button",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <li
                      className="menuitem"
                    >
                      <NavLink
                        activeClassName="active"
                        ariaCurrent="true"
                        className="menulink"
                        exact={true}
                        to="/"
                      >
                        Latest
                      </NavLink>
                    </li>,
                    <li
                      className="menuitem"
                    >
                      <NavLink
                        activeClassName="active"
                        ariaCurrent="true"
                        className="menulink"
                        to="/genres"
                      >
                        Genres
                      </NavLink>
                    </li>,
                    <li
                      className="menuitem"
                    >
                      <NavLink
                        activeClassName="active"
                        ariaCurrent="true"
                        className="menulink"
                        to="/about"
                      >
                        About
                      </NavLink>
                    </li>,
                  ],
                  "className": "menu",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": <NavLink
                        activeClassName="active"
                        ariaCurrent="true"
                        className="menulink"
                        exact={true}
                        to="/"
                      >
                        Latest
                      </NavLink>,
                      "className": "menuitem",
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "function",
                      "props": Object {
                        "activeClassName": "active",
                        "ariaCurrent": "true",
                        "children": "Latest",
                        "className": "menulink",
                        "exact": true,
                        "to": "/",
                      },
                      "ref": null,
                      "rendered": "Latest",
                      "type": [Function],
                    },
                    "type": "li",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": <NavLink
                        activeClassName="active"
                        ariaCurrent="true"
                        className="menulink"
                        to="/genres"
                      >
                        Genres
                      </NavLink>,
                      "className": "menuitem",
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "function",
                      "props": Object {
                        "activeClassName": "active",
                        "ariaCurrent": "true",
                        "children": "Genres",
                        "className": "menulink",
                        "to": "/genres",
                      },
                      "ref": null,
                      "rendered": "Genres",
                      "type": [Function],
                    },
                    "type": "li",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": <NavLink
                        activeClassName="active"
                        ariaCurrent="true"
                        className="menulink"
                        to="/about"
                      >
                        About
                      </NavLink>,
                      "className": "menuitem",
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "function",
                      "props": Object {
                        "activeClassName": "active",
                        "ariaCurrent": "true",
                        "children": "About",
                        "className": "menulink",
                        "to": "/about",
                      },
                      "ref": null,
                      "rendered": "About",
                      "type": [Function],
                    },
                    "type": "li",
                  },
                ],
                "type": "ul",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "search",
                  "className": "button material-icons",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": "search",
                "type": "button",
              },
            ],
            "type": "div",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "collapsed": false,
            "toggleSearchBox": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
